### PR TITLE
Add constant padding mode to the `torch.nn.functional.grid_sample` function

### LINKED ISSUE #111538: Propose to add constant padding mode to the `torch.nn.functional.grid_sample` function
### 🚀 The feature, motivation and pitch

I'm working on a registration project. When I am using `torch.nn.functional.grid_sample` to apply the displacement field to the moving image, I find I am unable to use constant values like 255 to pad it (at least it is not that intuitive) just like the way in `np.pad`. So I propose to add one. If possible I would really want to implement it myself, though I am a completely new contributor to the pytorch project.

### Alternatives

As a workaround, I pre-pad the image with 255 pixels on the border and use the `border` mode.

### Additional context

Here are two samples,
`zeros` mode
pre-pad `border` mode

### PR DESCRIPTION
**Pull request by TianyiFranklinWang**
Add constant padding mode to the `torch.nn.functional.grid_sample` function

Fixes #111538

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by TianyiFranklinWang** [comment]
Corresponding tests shall be added later.

**Comment by TianyiFranklinWang** [comment]
Hi @colesbury, kindly ask if anyone can give it a comment.

**Review by mikaylagawarecki — COMMENTED** [review]
Seems like tests are failing,

- you might need to update `torch/overrides.py` [here](https://github.com/pytorch/pytorch/blob/main/torch//overrides.py#L663-L665)
- I don't see any testing for the new `constant` mode that has been added, please add testing. You might also need to add some new sample inputs that exercise this [here](https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_methods_invocations.py?fbclid=IwZXh0bgNhZW0CMTEAAR3z4XGM092F4dYfV0myaBvyB0qaeqkO396veGb8QR4SZ9mseoY2rc2LYVE_aem_GF1PrugE71ngvAMKty2Liw#L20867) as well as testing [here](https://github.com/pytorch/pytorch/blob/main/test/test_nn.py?fbclid=IwZXh0bgNhZW0CMTEAAR2TTH1tJRijBKg4RviNmP4aHJlnTC0DY1jH7jEavZjyn3TuAPdw78MWjGE_aem_Vu8IwZQUu65IRRhQCvRGRQ#L5576)
- update the decompositions [here](https://github.com/pytorch/pytorch/blob/1962f9475fbc289eee35c1fc24bf71e43f5dbba8/torch/_decomp/decompositions.py#L634-L650)

**Comment by TianyiFranklinWang** [comment]
> Seems like tests are failing,
> 
> * you might need to update `torch/overrides.py` [here](https://github.com/pytorch/pytorch/blob/main/torch//overrides.py#L663-L665)
> * I don't see any testing for the new `constant` mode that has been added, please add testing. You might also need to add some new sample inputs that exercise this [here](https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_methods_invocations.py?fbclid=IwZXh0bgNhZW0CMTEAAR3z4XGM092F4dYfV0myaBvyB0qaeqkO396veGb8QR4SZ9mseoY2rc2LYVE_aem_GF1PrugE71ngvAMKty2Liw#L20867) as well as testing [here](https://github.com/pytorch/pytorch/blob/main/test/test_nn.py?fbclid=IwZXh0bgNhZW0CMTEAAR2TTH1tJRijBKg4RviNmP4aHJlnTC0DY1jH7jEavZjyn3TuAPdw78MWjGE_aem_Vu8IwZQUu65IRRhQCvRGRQ#L5576)
> * update the decompositions [here](https://github.com/pytorch/pytorch/blob/1962f9475fbc289eee35c1fc24bf71e43f5dbba8/torch/_decomp/decompositions.py#L634-L650)

Thank you for your time and review, I will start to fix these issues.
